### PR TITLE
Ahti production release readiness

### PR DESCRIPTION
**Pull request by ahie**

### CODE REVIEW

File: src/common/ui-components/CategoryIcon/CategoryIcon.tsx [hunk 1]
```diff
@@ -8,49 +8,35 @@ import { ReactComponent as IconWaterTaxi } from '../../../assets/icons/icon-wate
...
+};
 
-  return <Icon className={className ? className : styles.icon} />;
+const CategoryIcon: React.FC<CategoryIconProps> = ({ className, category }) => {
+  const Icon = CATEGORY_ICONS_MAP[category];
+  return Icon ? (
+    <Icon className={className ? className : styles.icon} />
+  ) : (
+    <IconLocation className={className ? className : styles.icon} />
+  );
```
Comment (Nizar-Rahme) [inline src/common/ui-components/CategoryIcon/CategoryIcon.tsx:39]:
```suggestion
  const Icon = CATEGORY_ICONS_MAP[category] || IconLocation; 
  return <Icon className={className ? className : styles.icon} />
```

File: src/common/ui-components/Map/PinIcon/PinIcon.tsx [hunk 2]
```diff
@@ -8,49 +8,49 @@ import { ReactComponent as IconWaterTaxi } from '../../../../assets/icons/pins/p
...
+const PinIcon: React.FC<CategoryIconProps> = ({ className, tags }) => {
+  if (!tags) {
+    return <IconLocation className={classNames(className, styles.offset)} />;
+  }
+  const Icon = PIN_ICONS_MAP[tags.find((tag) => PIN_ICONS_MAP[tag])];
+  return Icon ? (
+    <Icon className={classNames(className, styles.offset)} />
+  ) : (
+    <IconLocation className={classNames(className, styles.offset)} />
+  );
```
Comment (Nizar-Rahme) [inline src/common/ui-components/Map/PinIcon/PinIcon.tsx:53]:
```suggestion
  if (!tags) {
    return <IconLocation className={classNames(className, styles.offset)} />;
  }
  const Icon = PIN_ICONS_MAP[tags.find((tag) => PIN_ICONS_MAP[tag])] || IconLocation;
  return <Icon className={classNames(className, styles.offset)} />
```

File: src/domain/Footer/Footer.tsx [hunk 3]
```diff
@@ -41,6 +41,13 @@ const Footer: React.FC<FooterProps> = ({ className }) => {
         >
           {t('footer.sea_rescue')}
         </a>
+        <a
+          className={styles.link}
+          href={t('footer.feedback_url')}
+          target={'_blank'}
```
Comment (Nizar-Rahme) [inline src/domain/Footer/Footer.tsx:47]: https://searchenginelaws.com/seo/what-is-rel-noopener-noreferrer-tag/
```suggestion
          target="_blank"
          rel="noopener noreferrer"
```

File: src/domain/SuggestionPage/fields/TagSelect.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        {selectedTagIds.length > 0 && (
+          <Breadcrumb
+            items={selectedTagIds.map((selectedTagId) => {
+              return {
+                id: selectedTagId,
+              };
+            })}
+            onClose={onDeselect}
+            translations={tagAndCategoryTranslations}
+          />
```
Comment (Nizar-Rahme) [inline src/domain/SuggestionPage/fields/TagSelect.tsx:57]:
```suggestion
        {selectedTagIds.length === 0 ? (
          <span className={styles.noSelectedTags}>{noSelectedTagsText}</span>
        ) :
        (
          <Breadcrumb
            items={selectedTagIds.map((selectedTagId) => ({ id: selectedTagId })
            onClose={onDeselect}
            translations={tagAndCategoryTranslations}
          />
```

File: package.json [hunk 5]
```diff
@@ -49,9 +49,15 @@
...
+    "@types/yup": "^0.28.3",
     "@typescript-eslint/eslint-plugin": "^2.21.0",
     "@typescript-eslint/parser": "^2.21.0",
-    "apollo-boost": "^0.4.7",
+    "apollo-cache-inmemory": "^1.6.6",
+    "apollo-client": "^2.6.10",
+    "apollo-link": "^1.2.14",
+    "apollo-link-error": "^1.1.13",
+    "apollo-link-http": "^1.5.17",
+    "apollo-link-retry": "^2.2.16",
```
Comment (Nizar-Rahme) [inline package.json:65]: I'm interested in knowing the benefits of replacing `apollo-boost`.

File: src/common/ui-components/Select/Select.tsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+}) => {
+  const optionsItems = options.map(({ value, label }) => (
+    <option key={value} value={value}>
+      {label}
+    </option>
+  ));
+
+  return (
+    <div className={className}>
+      {labelText && <label className={styles.labelText}>{labelText}</label>}
```
Comment (Nizar-Rahme) [inline src/common/ui-components/Select/Select.tsx:42]:
```suggestion
      {labelText && <label className={styles.labelText}>{labelText}</label>}
```
```suggestion
      {labelText && <label className={styles.labelText} htmlFor={id}>{labelText}</label>}
```

File: src/domain/SuggestionPage/fields/TranslatedFieldInput.tsx [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+  labelText,
+  id,
+  value,
+  handleChange,
+  textArea,
+  error,
+}) => {
+  const Input = textArea ? TextArea : TextInput;
+  return (
+    <div>
```
Comment (Nizar-Rahme) [inline src/domain/SuggestionPage/fields/TranslatedFieldInput.tsx:28]:
```suggestion
    <>
```

File: src/common/translation/i18n/translations/fi.json [hunk 8]
```diff
@@ -217,17 +214,49 @@
...
+    "coordinates": "Valitse kartalta palvelun sijainti",
+    "selected_tags": "Valitut tagit",
+    "no_selected_tags": "Ei yhtäkään valittua tagia",
+    "on_submit": "Ehdotuksesi on vastaanotettu!",
+    "errors": {
+      "required": "Tämä kenttä on pakollinen",
+      "email": "Kentän arvo ei ole kelvollinen sähköposti",
+      "translated_field_required": "Suomalainen käännös tälle kentälle on pakollinen",
+      "check_form": "Tietoja ei voitu lähettä. Tarkista syöttämäsi tiedot."
+    }
```
Comment (Nizar-Rahme) [inline src/common/translation/i18n/translations/fi.json:260]: Swedish translations don't seem to be added.

File: src/common/ui-components/CategoryIcon/CategoryIcon.tsx (hunk 1 above)
Comment (ahie) [reply src/common/ui-components/CategoryIcon/CategoryIcon.tsx]: Done.

File: src/common/ui-components/Map/PinIcon/PinIcon.tsx (hunk 2 above)
Comment (ahie) [reply src/common/ui-components/Map/PinIcon/PinIcon.tsx]: Done.

File: src/domain/Footer/Footer.tsx (hunk 3 above)
Comment (ahie) [reply src/domain/Footer/Footer.tsx]: Done.

File: src/domain/SuggestionPage/fields/TagSelect.tsx (hunk 4 above)
Comment (ahie) [reply src/domain/SuggestionPage/fields/TagSelect.tsx]: Done.

File: package.json (hunk 5 above)
Comment (ahie) [reply package.json:65]: It gives more control to how you can configure the `ApolloClient` instance. Needed here for the custom links.

File: src/common/ui-components/Select/Select.tsx (hunk 6 above)
Comment (ahie) [reply src/common/ui-components/Select/Select.tsx]: Done.

File: src/domain/SuggestionPage/fields/TranslatedFieldInput.tsx (hunk 7 above)
Comment (ahie) [reply src/domain/SuggestionPage/fields/TranslatedFieldInput.tsx]: Changed to include a custom `className` instead to improve reuse.

File: src/common/translation/i18n/translations/fi.json (hunk 8 above)
Comment (ahie) [reply src/common/translation/i18n/translations/fi.json:260]: Swedish translations are still WIP. Will probably not make it into this PR.

File: src/domain/Footer/Footer.tsx (hunk 3 above)
Comment (Nizar-Rahme) [reply src/domain/Footer/Footer.tsx]: These links are still missing `rel="noopener noreferrer`

Comment (ahie) [reply src/domain/Footer/Footer.tsx]: Good catch. Also added these to `CardLinks` which was still missing them.

**Pull request merged**